contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -2566,11 +2568,40 @@ tsql_coerce_string_literal_hook(Oid targetTypeId,
 		 */
 		for (i = strlen(value) - 1; i >= 0; i--)
 		{
-			if (value[i] != ' ')
+			if (!isspace((unsigned char)value[i]))
 				break;
 		}
 
-		if (i == -1)
+		if (i == -1 && targetTypeId == DATEOID)
+		{
+			/*
+			 * Set input to default '1900-01-01' if empty string encountered
+			 */
+			DateADT date;
+			struct pg_tm tt,
+						 *tm = &tt;
+			tm->tm_year = 1900;
+			tm->tm_mon = 1;
+			tm->tm_mday = 1;
+			date = date2j(tm->tm_year, tm->tm_mon, tm->tm_mday) - POSTGRES_EPOCH_JDATE;
+
+			newcon->constvalue = DateADTGetDatum(date);
```

Comment:
Arent all of these conditions internal logic duplicate to initializeToDefaultDate/initializeToDefaultTime. If they are, kindly avoid duplication and please use the functions directly

Reply:
Thats not the right way of importing between extensions. Please check this out: https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_5_X_DEV/contrib/babelfishpg_common/src/babelfishpg_common.h#L38
If needed, we can have a discussion about it offline.